File: Backend/fastApiProject/model/location.py
from sqlalchemy import Column, Integer, Text, Numeric, DateTime, func
from geoalchemy2 import Geography
from config.database import Base


class Location(Base):
    __tablename__ = 'locations'

    id = Column(Integer, primary_key=True)
    latitude = Column(Numeric, nullable=False)
    longitude = Column(Numeric, nullable=False)
    city = Column(Text, nullable=True)
    country = Column(Text, nullable=True)
    coordinates = Column(Geography(geometry_type="POINT", srid=4326), nullable=False)
    created_at = Column(DateTime, nullable=False, server_default=func.now())
    updated_at = Column(DateTime, nullable=False, server_default=func.now(), onupdate=func.now())

File: disaster-live-alerts/src/main/java/com/example/disaster_live_alerts/model/Location.java
package com.example.disaster_live_alerts.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import org.locationtech.jts.geom.Point;

import java.time.LocalDateTime;

@Entity
@Table(name = "location")
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Location {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    private String city;

    private String country;

    @Column(name = "coordinates", columnDefinition = "geography(Point, 4326)")
    private Point coordinates;

    @Builder.Default
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Builder.Default
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();
}
